sdk/python/kfp/compiler/compiler.py

```diff
@@ -55,6 +55,19 @@ def my_pipeline(a: int = 1, b: str = "default value"):
       Compiler().compile(my_pipeline, 'path/to/workflow.yaml')
   """
 
+  def __init__(
+      self,
+      mode: dsl.PipelineExecutionMode = dsl.PipelineExecutionMode.V1_LEGACY):
+    if mode == dsl.PipelineExecutionMode.V2_ENGINE:
+      raise ValueError('V2_ENGINE execution mode is not supported yet.')
+
+    if mode == dsl.PipelineExecutionMode.V2_COMPATIBLE:
+      warnings.warn('V2_COMPATIBLE execution mode is still under development.'
+                    ' Pipelines may not work as expected.')
+    self._mode = mode
+    self._pipeline_name = None
+    self._pipeline_root = None
```

Comment:
nit: the name of `self._pipeline_name` and `self._pipeline_root` make me think they are string typed values. But later I realize that they are actually `dsl.PipelineParam` typed objects. How about renaming them to something like `self._pipeline_name_param` and `self._pipeline_root_param`?

In addition to renaming, optionally they can be type declared as, for example:
`self._pipeline_name_param: dsl.PipelineParam = None`
`self._pipeline_root_param: dsl.PipelineParam = None`

Reply:
I annotated the type, and after that the IDE tends to show the type of the variable. Will this suffice or do you still prefer the `_param` suffix?

Reply:
I kinda prefer having the `_param` suffix anyway, it helps when reading the code without advanced IDEs -- like browsing the code on GitHub.
Right now in `_create_workflow` function, there's a local variable named `pipeline_name` which is string typed, and there's the instance variable `self._pipeline_name` that is a `dsl.PipelineParam`. The two very similar names having different types hurt readability IMO. WDYT?

Reply:
Done.